classes/DepositPointsHelper.inc.php

```diff
@@ -23,16 +23,22 @@ class DepositPointsHelper {
 	 * @param $password string
 	 * @return array|null
 	 */
-	public static function loadCollectionsFromServer($url, $username, $password) {
-		if (empty($url) OR empty($username) OR empty($password)) {
+	public static function loadCollectionsFromServer($url, $username, $password, $apikey) {
+		if (empty($url) OR (empty($username) AND empty($password)) OR empty($apikey)) {
 			return null;
 		}
+		if ($apikey) {
+			$url = $url . "?apiToken=" . $apikey;
+		}
 		$depositPoints = array();
 		$client = new SWORDAPPClient();
 		$doc = $client->servicedocument($url, $username, $password, '');
```

Comment:
Have you checked the SWORD protocol for opinions on the `apiKey` parameter? I don't know about SWORD, but OAI-PMH for example forbids the use of additional URL parameters beyond those described in the spec.

Reply:
I don't find anything in the technical documentation beyond the requirement that a SWORD server support HTTP basic auth. My sense is that the standard is unopinionated as to any additional authorization schemes. It does look to me from the mail list like other implementers have felt free to add what they need: https://sourceforge.net/p/sword-app/mailman/message/31891842/

Reply:
Perhaps it makes sense to use a request header rather than a URL parameter to carry the apikey? I don't think there's any downside to doing that.

Reply:
As it turns out, I think it isn't going to be possible to do this without forking the sword client library, due to this line:
https://github.com/swordapp/swordappv2-php-library/blob/46e2dbb86611d4b44880e62301cbec49cb9549cc/swordappclient.php#L33
I can fork it and send a PR - not sure if it is still maintained.

Reply:
https://github.com/swordapp/swordappv2-php-library/pull/13/files